ts/modules/full-screen.src.ts

```
@@ -18,63 +15,194 @@ import H from '../parts/Globals.js';
+H.Chart.prototype.toggleFullscreen = function (): void {
+    const chart = this;
+
+    if (!(chart.container.parentNode instanceof HTMLElement)) {
+        return;
+    }
+
+    const container = chart.container.parentNode;
+
+    // Hold event and methods available only for a current browser.
+    if (!chart.browserProps) {
+        if (typeof container.requestFullscreen === 'function') {
+            chart.browserProps = {
+                fullscreenChange: 'fullscreenchange',
+                requestFullscreen: 'requestFullscreen',
+                exitFullscreen: 'exitFullscreen'
+            };
         } else if (container.mozRequestFullScreen) {
-            promise = container.mozRequestFullScreen();
-        } else if (container.webkitRequestFullscreen) {
-            promise = container.webkitRequestFullscreen();
+            chart.browserProps = {
+                fullscreenChange: 'mozfullscreenchange',
+                requestFullscreen: 'mozRequestFullScreen',
+                exitFullscreen: 'mozCancelFullScreen'
+            };
+        } else if (container.webkitRequestFullScreen) {
+            chart.browserProps = {
+                fullscreenChange: 'webkitfullscreenchange',
+                requestFullscreen: 'webkitRequestFullScreen',
+                exitFullscreen: 'webkitExitFullscreen'
+            };
         } else if (container.msRequestFullscreen) {
-            promise = container.msRequestFullscreen();
+            chart.browserProps = {
+                fullscreenChange: 'MSFullscreenChange',
+                requestFullscreen: 'msRequestFullscreen',
+                exitFullscreen: 'msExitFullscreen'
+            };
         }
+    }
 
-        if (promise) {
-            promise['catch']((): void => { // eslint-disable-line dot-notation
-                alert('Full screen is not supported inside a frame'); // eslint-disable-line no-alert
-            });
-        }
+    if (!chart.isFullscreen) {
+        chart.openFullscreen();
```

The error message is now removed. Previously I would get an alert when trying to full-screen a chart from inside a frame, now it just silently fails but with a JS error in the console. This can be tested in the utils.